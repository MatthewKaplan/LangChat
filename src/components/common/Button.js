import React from 'react';
import { Text, TouchableOpacity } from 'react-native';

const Button = ({ onPress, children, disabled }) => {
	const { buttonStyle, textStyle } = styles;
	return (
		<TouchableOpacity onPress={onPress} style={buttonStyle} disabled={disabled}>
			<Text style={textStyle}>{children}</Text>
		</TouchableOpacity>
	);
};

const styles = {
	textStyle: {
		alignSelf: 'center',
		color: '#fff',
		fontSize: 19,
		fontWeight: '600',
		paddingTop: 12,
		paddingBottom: 12
	},
	buttonStyle: {
		position: 'absolute',
		alignSelf: 'stretch',
		width: '90%',
		backgroundColor: '#007aff',
		borderRadius: 5,
		borderWidth: 1,
		borderColor: '#007aff',
		marginTop: 20,
		marginLeft: 5,
		marginRight: 5,
		alignSelf: 'center',
		top: '84%'
	}
};

export default Button;
