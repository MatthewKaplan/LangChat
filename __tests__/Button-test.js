// __tests__/Button-test.js
import React from 'react';
import Button from '../src/components/common/Button';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
	const tree = renderer.create(<Button />).toJSON();
	expect(tree).toMatchSnapshot();
});
