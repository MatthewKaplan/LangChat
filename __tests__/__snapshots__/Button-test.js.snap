// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#007aff",
      "borderRadius": 5,
      "margin": 10,
      "opacity": 1,
      "width": "90%",
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#fff",
        "fontSize": 19,
        "fontWeight": "600",
        "paddingBottom": 12,
        "paddingTop": 12,
      }
    }
  />
</View>
`;
